cclient/api/extended/get_inputs.c

```diff
@@ -4,63 +4,130 @@
+                                   bool* const is_unused) {
   retcode_t ret_code = RC_ERROR;
-  hash243_queue_entry_t* q_iter = NULL;
-  size_t counter = 0;
-  get_balances_req_t* balances_req = get_balances_req_new();
-  get_balances_res_t* balances_res = get_balances_res_new();
+  size_t ret_num = 0;
+  find_transactions_req_t* find_tran_req = NULL;
+  find_transactions_res_t* find_tran_res = NULL;
 
   log_debug(client_extended_logger_id, "[%s:%d]\n", __func__, __LINE__);
-  if (!balances_req || !balances_res) {
+  find_tran_req = find_transactions_req_new();
+  find_tran_res = find_transactions_res_new();
+  if (!find_tran_req || !find_tran_res) {
     ret_code = RC_OOM;
-    log_error(client_extended_logger_id, "%s create get balances request or response object failed: %s\n", __func__,
+    log_error(client_extended_logger_id, "%s: create find transactions request or response failed: %s\n", __func__,
               error_2_string(ret_code));
-    return ret_code;
+    goto done;
   }
 
-  out_inputs->total_balance = 0;
-
-  // get address list
-  ret_code = iota_client_get_new_address(serv, seed, addr_opt, &balances_req->addresses);
+  ret_code = hash243_queue_push(&find_tran_req->addresses, addr);
   if (ret_code) {
-    log_error(client_extended_logger_id, "%s get new address failed: %s\n", __func__, error_2_string(ret_code));
+    log_error(client_extended_logger_id, "%s: hahs queue push failed: %s\n", __func__, error_2_string(ret_code));
     goto done;
   }
+  ret_code = iota_client_find_transactions(serv, find_tran_req, find_tran_res);
+  if (ret_code == RC_OK) {
+    ret_num = hash243_queue_count(find_tran_res->hashes);
+    *is_unused = ret_num ? false : true;
+  }
+done:
+  find_transactions_req_free(&find_tran_req);
+  find_transactions_res_free(&find_tran_res);
+  return ret_code;
+}
+
+retcode_t iota_client_get_inputs(iota_client_service_t const* const serv, flex_trit_t const* const seed,
+                                 address_opt_t const addr_opt, uint64_t const threshold, inputs_t* const out_inputs) {
+  retcode_t ret_code = RC_ERROR;
+  get_balances_req_t* balances_req = NULL;
+  get_balances_res_t* balances_res = NULL;
 
-  balances_req->threshold = 100;  // currently `100` should be used
-  ret_code = iota_client_get_balances(serv, balances_req, balances_res);
-  if (ret_code == RC_OK && balances_req->addresses) {
-    // expect balance value is in order.
-    input_t tmp_input = {};
-    CDL_FOREACH(balances_req->addresses, q_iter) {
-      if (out_inputs->total_balance < (int64_t)threshold) {
-        tmp_input.balance = get_balances_res_balances_at(balances_res, counter);
-        tmp_input.key_index = counter;
+  out_inputs->total_balance = 0;
+
+  flex_trit_t* addr = NULL;
+  bool is_unused = false;
+  uint64_t addr_index, addr_index_end;
+  if (addr_opt.total == 0) {
+    addr_index = 0;
+    addr_index_end = (unsigned)-1;
```

Comment:
`addr_index_end` is `uint64_t`

Reply:
I want to assign maximun possible value of `uint64_t` to `addr_index_end`. Use an `(unsigned)-1` would be a simple way without including other header file

Reply:
the maximum value of `uint64_t`(18446744073709551615) is no equal to `unsigned int -1`(4294967295)